src/H5trace.c

```diff
@@ -3722,5 +3722,5 @@ H5_trace(const double *returning, const char *func, const char *type, ...)
     if (H5_debug_g.ttimes)
         return function_times.elapsed;
     else
-        return 0.0F;
+        return (double)0.0F;
```

Comment:
These casts to suffixed floats (cruft leftover from vainly trying to deal with gcc's insistence that we should use their non-standard D suffix) need to go.